cluster-autoscaler/utils.go

```diff
@@ -271,3 +271,15 @@ func fixNodeGroupSize(context *AutoscalingContext, currentTime time.Time) (bool,
 	}
 	return fixed, nil
 }
+
+// Enables cloud provider specific node protection for each node group.
+func protectNodes(context *AutoscalingContext) error {
+	for _, group := range context.CloudProvider.NodeGroups() {
```

Comment:
so maybe we should name it in a different - "protection" indicates an action that is specific to a cloud provider. If we don't go with the full nodegroup protection then I would suggest calling this PrepareNodeGroup or similar so that if a different cloud provider will need some other action executed on each of the node groups we won't be adding the next method.